cumulus/client/cirrus-executor/src/bundle_processor.rs

```diff
@@ -472,35 +472,50 @@ where
 			.extract_receipts(&BlockId::Hash(primary_hash), extrinsics.clone())?;
 
 		for signed_receipt in signed_receipts.iter() {
+			let secondary_hash = signed_receipt.execution_receipt.secondary_hash;
 			match crate::aux_schema::load_execution_receipt::<
 				_,
 				Block::Hash,
 				NumberFor<PBlock>,
 				PBlock::Hash,
-			>(&*self.client, signed_receipt.execution_receipt.secondary_hash)?
+			>(&*self.client, secondary_hash)?
 			{
 				Some(local_receipt) => {
 					if let Some(trace_mismatch_index) =
 						find_trace_mismatch(&local_receipt, &signed_receipt.execution_receipt)
 					{
-						crate::aux_schema::write_bad_receipt::<_, PBlock>(
+						let trace_mismatch_index = trace_mismatch_index
+							.try_into()
+							.expect("Trace mismatch index must fit into u32; qed");
```

Comment:
`usize` depends on the system you're running on. I would say we need to avoid `usize` when it comes to storing things on disk or sending over the network because it will potentially cause problems (not very likely due to most systems being 64-bit, but still). This is probably why SCALE codec is not implemented for `usize`.

Please switch back to `u32` there.

Reply:
So now we're back to my original comment: why `find_trace_mismatch` returns `usize` and not `u32` and why `generate_proof` takes `usize` instead of `u32` too? If we expect length of execution trace to always fit into `u32` and design data structures around it, there is no point in having functions return or take anything else for trace index.

Reply:
I don't think we have any constraint on the execution trace on the spec(just ignore the limit from other factors like block size, network bandwidth, etc), that's why I don't want to oppose this limit forcibly in the implementation if we don't have to.

Reply:
Well, block size is absolutely a constraint on the block size. `u32` means that even if every transaction was 1 bytes, you'd have a block that is 4GiB in size! Even if it was possible and we needed more than `u32`, we'd have to use `u64` explicitly or those who compiled node for, let's say, 32-bit ARM platform will crash. So I see no reason to have `usize` here.

Reply:
Well, then you don't have to convert `u32` to `usize` in this comparison https://github.com/subspace/subspace/blob/9ab78aff3456a8e32feccbe6e5cea70a0db0012f/cumulus/client/cirrus-executor/src/fraud_proof.rs#L141

I don't think it technically matters as you basically won't expect an executor to run a 32-bit ARM platform. I can just update if you insist.

Reply:
We expect index to fit into `u32`, then it must be `u32`. If we expect it to be bigger than that we need to make it `u64` (but we don't). Consensus-critical stuff shouldn't depend on the address size of the platform it is compiled on.